src/fable/Fable.Compiler/Replacements.fs

```diff
@@ -400,17 +400,10 @@ module private AstPass =
                     | Some t -> Some(Fable.Array t)
                     | None -> None
                 | Fable.Tuple ts ->
-                    (ts, (false, []))
-                    ||> List.foldBack (fun t (inflate, acc) ->
-                        match needsInflate genArgsMap t with
-                        | Some t -> true, t::acc
-                        | None -> inflate, t::acc)
-                    |> function
-                    | true, ts -> Some(Fable.Tuple ts)
-                    | false, _ -> None
+                    Some(Fable.Tuple ts)
                 | Fable.DeclaredType(ent, [genArg])
                     when ent.FullName = "Microsoft.FSharp.Core.FSharpOption" ->
-                    needsInflate genArgsMap genArg
+                    Some(Fable.DeclaredType(ent, [genArg]))
```

Comment:
We need to go into the tuple to extract/output the length of the tuple as it's not just an array, also mark it as such so we can ```if (expected === "Tuple")``` later on.

As options in json.net are not just simple values, we need to output that it's an FSharpOption.  We could just check for ```Case == "Some"``` but I don’t think that is as safe